src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -43,18 +43,17 @@ internal BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, Dia
                     case SyntaxKind.ExpressionStatement:
                         if (expression != null)
                         {
-                            // We only allow assignment-only or declaration-only deconstructions at this point.
-                            // Issue https://github.com/dotnet/roslyn/issues/15050 tracks allowing mixed deconstructions.
-                            // For now we give an error when you mix.
-                            Error(diagnostics, ErrorCode.ERR_MixedDeconstructionUnsupported, left);
+                            MessageID.IDS_FeatureMixedDeclarationsAndExpressionsInDeconstructor
```

Comment:
`Deconstructor` -> `Deconstruction`. I don't think we use the term `deconstructor` anywhere.